openlcs/libs/common.py

```diff
@@ -287,6 +288,30 @@ def get_extension(file_name, sp_extensions):
     return os.path.splitext(file_name)
 
 
+def get_nvr_from_provide(provide):
+    """
+    Get nvr from parse the purl of the provide
+    """
+    purl = provide.get('purl')
+    purl = PackageURL.from_string(purl)
+    if purl.to_dict().get('namespace') and \
+            purl.to_dict().get('namespace') != 'redhat':
+        # Add the namespace to the nvr for remote source
```

Comment:
There is no need to deduplicate on a remote source, it has only one arch, the `noarch`.

Reply:
This function is just to get the 'nvr' from the 'provide. purl',  it won't be used for deduplication.  It's more simple to refine the 'provides' list when we re-create the 'provides' list according to the 'nvrs'. 
If we remove this remote source to the nvrs, so we need to do more in the function deduplicate_provides in openlcs/libs/corgi. We need to handle the oci_noarch_provides list according to the namespace(the remote source and the rpm).

Reply:
If remove getting nvr of remote source as your suggestion, here are the implementation:

def get_nvr_from_provide(provide):
    """
    Get rpm nvr from parsing the purl of the provide
    """
    purl = provide.get('purl')
    purl_dict = PackageURL.from_string(purl).to_dict()
    nvr = '-'.join((purl_dict.get('name'),
                        purl_dict.get('version')))
    return nvr
      
    
def deduplicate_provides(self, oci_noarch_provides):
        # Deduplicate provides for components with different arch.
        nvrs = []
        provides = []
        provides_rs = []
        provides_rpm = []
        for oci_noarch_provide in oci_noarch_provides:
            purl =  oci_noarch_provide.get('purl')
            purl_dict = PackageURL.from_string(purl).to_dict()
            if purl_dict.get('namespace') == 'redhat':
                nvr = get_nvr_from_provide(oci_noarch_provide)
                if nvr not in nvrs:
                    nvrs.append(nvr)
                    provides_rpm.append(oci_noarch_provide)
            else:
                provides_rs.append(oci_noarch_provide)
        provides = provides_rs provides_rpm
        return provides

Do you prefer this implementation?